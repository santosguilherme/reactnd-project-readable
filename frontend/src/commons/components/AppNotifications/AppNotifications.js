import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';

import Notifications from 'react-notification-system-redux';


function AppNotifications({notifications}) {
    const style = {
        NotificationItem: {
            DefaultStyle: {
                fontSize: '0.875rem',
                fontWeight: '400',
                fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
                lineHeight: '1.46429em'
            }
        }
    };

    return (
        <Notifications
            notifications={notifications}
            style={style}
        />
    );
}

AppNotifications.defaultProps = {
    notifications: []
};

AppNotifications.propTypes = {
    notifications: PropTypes.array
};

function mapStateToProps(state) {
    const {notifications} = state;
    return {notifications};
}

export default connect(mapStateToProps)(AppNotifications);
