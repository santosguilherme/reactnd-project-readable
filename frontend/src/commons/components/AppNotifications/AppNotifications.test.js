import React from 'react';
import {shallow} from 'enzyme';

import {AppNotifications} from './AppNotifications';


it('renders without crashing', () => {
    const wrapper = shallow(<AppNotifications/>);

    expect(wrapper).toMatchSnapshot();
});