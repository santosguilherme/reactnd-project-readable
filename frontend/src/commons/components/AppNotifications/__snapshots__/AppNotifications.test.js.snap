// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppNotifications
    notifications={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "notifications": Array [],
      "style": Object {
        "NotificationItem": Object {
          "DefaultStyle": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": "400",
            "lineHeight": "1.46429em",
          },
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "notifications": Array [],
        "style": Object {
          "NotificationItem": Object {
            "DefaultStyle": Object {
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "0.875rem",
              "fontWeight": "400",
              "lineHeight": "1.46429em",
            },
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
